pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+        ":FETC?",
+        """ Queries the most recent measurement data. """
+    )
+
+    #############
+    # Frequency #
+    #############
+    frequency = Instrument.measurement(
+        ":FREQ?",
+        """ Return the fundamental wave, primary frequency. """
+    )
+
+    harmonics_pre_enable = Instrument.control(
+        ":FREQ:HARM?", ":FREQ:HARM %d",
+        """ A property that enable/disable the harmonics measurement. """,
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    harmonics_pre_order = Instrument.control(
+        ":FREQ:MULT?", ":FREQ:HARM ON; :FREQ:MULT %d",
+        """ A property that control the harmonic order n for measurement (Primary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+    
+    subharmonic_order = Instrument.control(
+        "FREQ:SMUL?", ":FREQ:HARM ON; :FREQ:MULT %d",
+        """ A property that control the subharmonic order m for measurement (Primary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+
+    secondary_frequency = Instrument.measurement(
+        ":FREQ2?",
+        """ Return the secondary frequency used with detection modes DUAL2 and CASCADE. """
+    )
+
+    harmonics_sec_enable = Instrument.control(
+        ":FREQ2:HARM?",":FREQ2:HARM %d",
+        """ A property that control the harmonic measurement (enabled or disabled) (secondary PSD)""",
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    harmonics_sec_order = Instrument.control(
+        ":FREQ2:MULT?", ":FREQ2:HARM ON; :FREQ:MULT %d",
+        """ A property that control the harmonic order n for measurement (secondary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+
+    phase_shift = Instrument.control(
+        ":PHAS?", ":PHAS %g",
+        """ A property that control the phase shift amount (primary PSD). """,
+        validator=truncated_range,
+        values=[-180.000, 179.000]
+    )
+
+    ##########
+    # Source #
+    ##########
+    source_frequency = Instrument.control(
+        ":SOUR:FREQ?", "SOUR:FREQ %d",
+        """ A property that control the internal oscillator (primary PSD) frequency. """,
+        validator=truncated_range,
+        values=[300e-3, 1.15e+7]
+    )
+
+    source_voltage = Instrument.control(
+        ":SOUR:VOLT?", ":SOUR:VOLT %d",
+        """ A property that control the internal oscillator output voltage AC amplitude. """,
+        validator=truncated_range,
+        values=[0, 1]
+    )
+
+    ############
+    # Methods  #
+    ############
+    def default_display_screen(self):
+        """ Configures the measurement screen. """
+        self.display_screen = "Fine"
+
+    def default_dynamic_reserve(self):
+        """ Configure the dynamic reserve Sensitivity. """
+        self.dynamic_reserve = "Medium"
+
+    def scrn(self, DISPLAY=None):
```

Comment:
Please don't use ALL_CAPS parameter names, here use `display` insteady (similar in other method definitions)

Reply:
Thanks, all changed